librz/bin/dwarf.c

```diff
@@ -358,55 +484,86 @@ RZ_API const char *rz_bin_dwarf_get_attr_name(ut64 attr_code) {
 	}
 }
 
-RZ_API const char *rz_bin_dwarf_get_attr_form_name(ut64 form_code) {
+RZ_API const char *rz_bin_dwarf_form(enum DW_FORM form_code) {
 	if (form_code < DW_FORM_addr || form_code > DW_FORM_addrx4) {
 		return NULL;
 	}
 	return dwarf_attr_form_encodings[form_code];
 }
 
-RZ_API const char *rz_bin_dwarf_get_unit_type_name(ut64 unit_type) {
+RZ_API const char *rz_bin_dwarf_unit_type(enum DW_UT unit_type) {
 	if (!unit_type || unit_type > DW_UT_split_type) {
 		return NULL;
 	}
 	return dwarf_unit_types[unit_type];
 }
 
-RZ_API const char *rz_bin_dwarf_get_lang_name(ut64 lang) {
+RZ_API const char *rz_bin_dwarf_lang(enum DW_LANG lang) {
 	if (lang >= RZ_ARRAY_SIZE(dwarf_langs)) {
 		return NULL;
 	}
 	return dwarf_langs[lang];
 }
 
-static int abbrev_cmp(const void *a, const void *b) {
-	const RzBinDwarfAbbrevDecl *first = a;
-	const RzBinDwarfAbbrevDecl *second = b;
+static const char *dwarf_children[] = {
+	[DW_CHILDREN_yes] = "DW_CHILDREN_yes",
+	[DW_CHILDREN_no] = "DW_CHILDREN_no",
+};
 
-	if (first->offset > second->offset) {
-		return 1;
-	} else if (first->offset < second->offset) {
-		return -1;
-	} else {
-		return 0;
+RZ_API const char *rz_bin_dwarf_children(enum DW_CHILDREN lang) {
+	if (lang >= RZ_ARRAY_SIZE(dwarf_children)) {
+		return NULL;
 	}
+	return dwarf_children[lang];
 }
 
 /**
  * \brief Read an "initial length" value, as specified by dwarf.
  * This also determines whether it is 64bit or 32bit and reads 4 or 12 bytes respectively.
  */
 static inline ut64 dwarf_read_initial_length(RZ_OUT bool *is_64bit, bool big_endian, const ut8 **buf, const ut8 *buf_end) {
+	static const ut64 DWARF32_UNIT_LENGTH_MAX = 0xfffffff0;
```

Comment:
I see these are used multiple times, maybe move them to the top of the file and define only once?

Reply:
I haven't finished refactoring yet, and actually this function is going to be deleted in the end.